.github/workflows/spark_master_test.yaml

```diff
@@ -46,7 +48,7 @@ jobs:
       - name: Run Spark Master tests
         # when changing TEST_PARALLELISM_COUNT make sure to also change it in spark_test.yaml
         run: |
-          TEST_PARALLELISM_COUNT=2 build/sbt -DsparkVersion=master "++ ${{ matrix.scala }}" clean spark/test
+          TEST_PARALLELISM_COUNT=2 SHARD_ID=${{matrix.shard}} build/sbt -DsparkVersion=master "++ ${{ matrix.scala }}" clean spark/test
```

Comment:
how does this select just a single shard integer? To me this looks like it would select the entire array

Reply:
The GH matrix option will tell GH actions to trigger a job per matrix combination. https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-a-matrix-for-your-jobs

So in this case for spark master there will be a  job for Scala 2.13.13 for shard 0, another one for 1, for 2 etc.

matrix.{value} always references what is the concrete value for that given job. 
For example, It's how the delta spark tests work for testing the different scala versions (2.12.8 and 2.13.13 in spark_test.yaml). Those scala versions ultimately get passed through to `run-tests.py`